core/drivers/semihosting_console.c

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024 Andes Technology Corporation
+ */
+
+#include <compiler.h>
+#include <drivers/semihosting_console.h>
+#include <kernel/semihosting.h>
+#include <util.h>
+
+static void semihosting_console_putc(struct serial_chip *chip __unused, int ch)
+{
+	semihosting_writec(ch);
+}
+
+static int semihosting_console_getchar(struct serial_chip *chip __unused)
+{
+	return semihosting_readc();
+}
+
+static const struct serial_ops semihosting_console_ops = {
+	.putc = semihosting_console_putc,
+	.getchar = semihosting_console_getchar,
+};
+DECLARE_KEEP_PAGER(semihosting_console_ops);
+
+static void semihosting_console_fd_putc(struct serial_chip *chip __unused,
```

Comment:
remove `__unused`

Reply:
Addressed in https://github.com/OP-TEE/optee_os/pull/6700/commits/c036e76844763d61d6abe980e9b7ac880993cd5d